site-src/geps/gep-957.md

```diff
@@ -0,0 +1,77 @@
+# GEP-957: Destination Port Matching
+
+* Issue: [#957](https://github.com/kubernetes-sigs/gateway-api/issues/957)
+* Status: Provisional
+
+## TLDR
+
+Support destination port matching.
+
+## Goals
+
+* Support destination port matching based on port number.
+
+## Non-Goals
+
+* Support port matching based on named port.
+
+## Introduction
+
+Port matching is a common service mesh use case where traffic policies/rules
+need to be applied to traffic to certain destination ports. For ingress, while
+the API today already supports attaching a route to a specific listener, it may
+be useful to support attaching routes to listener(s) on a specified port. For
+example, some users may only want to expose their application on ports 80 or 443
+on the Gateway.
+
+## API
```

Comment:
Probably need a section in the GEP and/or godoc below describing the interaction between SectionName and Port. I'm not actually sure what the best approach is though.

We could say they were mutually exclusive, but that is likely unnecessarily restrictive. For ingress, one could say that they only wanted to attach to Gateway listener named foo if it was also listening on port 80. I'm assuming that wouldn't be super common, but it does seem fairly easy to interpret. For mesh, section name could represent some other implementation specific attribute such as a network name. In that case, it seems like specifying both network/sectionName and port could be quite helpful.

Given the potential usefulness, maybe we should suggest that these fields are complementary?

Reply:
Yeah, I don't have a strong opinion on this. Agreed that it may be better to make these fields complementary given that there is a reasonable interpretation on the meaning when both fields are present.